pkg/vrct/vrctFs/vrct.go

```diff
@@ -141,140 +172,299 @@ func mergePrototypes(prototypesDirPath, destPath string) error {
+		if err != nil {
+			return finalLayer, err
 		}
-		potentialConflictError := fmt.Errorf(
-			"%s layer nr. %d is in conflict with previous layers", optionalInfoPrefix, i,
-		)
 
-		pl.ConfigOptions = append(pl.ConfigOptions, layer.ConfigOptions...)
-		pl.IsOptional = layer.IsOptional && pl.IsOptional
+		finalLayerContent, finalLayerOptions, err = mergeConfigs(finalLayerContent, finalLayerOptions, currentLayerContent, currentLayerOptions, layer.IsOptional)
+		if err != nil {
+			return finalLayer, err
+		}
+	}
 
-		if pl.ConfigType != nil && layer.ConfigType != nil {
-			return pl, potentialConflictError
+	marshalledContent, err := bson.Marshal(finalLayerContent)
+	if err != nil {
+		return finalLayer, err
+	}
+	err = finalLayer.SetContent(marshalledContent)
+
+	err = SaveBsonMap(finalLayerOptions, finalLayer.OptionsPath)
+
+	return finalLayer, err
+}
+
+func getBoolMap(value interface{}, option interface{}) (map[string]interface{}, map[string]interface{}, error) {
+	valueKind := reflect.ValueOf(value).Kind()
+	optionKind := reflect.ValueOf(option).Kind()
+
+	if valueKind != reflect.Map {
+		return nil, nil, fmt.Errorf("trying to map interface that is not map")
+	}
+
+	mappedValue := value.(map[string]interface{})
+
+	var mappedOption map[string]interface{}
+	if optionKind == reflect.Map {
+		mappedOption = option.(map[string]interface{})
+	} else if option == reflect.Bool {
+		for key := range mappedValue {
+			mappedOption[key] = option.(bool)
 		}
-		pl.ConfigType = layer.ConfigType
+	} else {
+		return nil, nil, fmt.Errorf("types conflict")
 	}
 
-	// Not optional layers first
-	sort.Slice(p.Layers, func(i, j int) bool {
-		return !p.Layers[i].IsOptional && p.Layers[j].IsOptional
-	})
+	return mappedValue, mappedOption, nil
+}
 
-	for i, layer := range p.Layers {
-		if pl.ContentPath != nil && layer.ContentPath != nil && !layer.IsOptional {
-			return pl, fmt.Errorf(
-				"non optional layer nr. %d is in conflict with previous layers", i,
-			)
+func getBoolArray(value interface{}, option interface{}) ([]any, []any, error) {
+	valueKind := reflect.ValueOf(value).Kind()
+	optionKind := reflect.ValueOf(option).Kind()
+
+	if valueKind != reflect.Slice {
+		return nil, nil, fmt.Errorf("trying to map interface that is not map")
+	}
+	arrayedValue := value.([]any)
+
+	var arrayedOption []any
+	if optionKind == reflect.Slice {
+		arrayedOption = option.([]any)
+	} else if optionKind == reflect.Bool {
+		for range arrayedValue {
+			arrayedOption = append(arrayedOption, option.(bool))
+		}
+	} else {
+		return arrayedValue, nil, fmt.Errorf("options structure does not match config's one: types conflict '%s' and '%s'", valueKind, optionKind)
+	}
+
+	return arrayedValue, arrayedOption, nil
+}
+
+func mergeConfigs(merger map[string]interface{}, mergerOptions map[string]interface{}, toMerge map[string]interface{}, toMergeOptions map[string]interface{}, isOptional bool) (map[string]interface{}, map[string]interface{}, error) {
```

Comment:
```suggestion
func mergeConfigs(merger map[string]any, mergerOptions map[string]any, toMerge map[string]any, toMergeOptions map[string]any, isOptional bool) (map[string]any, map[string]any, error) {
```
It isn't good idea to write `interface{}` here, because this function declaration is already radically long